packages/rspack-cli/src/rspack-cli.ts

```diff
@@ -62,7 +57,9 @@ export class RspackCLI {
 			? (config as MultiRspackOptions).some(i => i.watch)
 			: (config as RspackOptions).watch;
 
-		return rspack(config, isWatch ? callback : undefined);
```

Comment:
Why using type casting here? It's a bad practice to me as it hinders the potential type changes.

Reply:
> Why using type casting here? It's a bad practice to me as it hinders the potential type changes.

Because if we remove this, the type will inferred be 
![image](https://github.com/web-infra-dev/rspack/assets/26035718/c7040db1-4bad-4110-8459-9048237b32ce)

Just MultiCompiler reserverd

Reply:
It does not seem to be that case for me.

Reply:
> It does not seem to be that case for me.

Maybe my IDE is webstorm, the difference may comes from the different ways in which two IDEs infer types, i will remove this.